### ISSUE TITLE
MBH Bombay edition links

### ISSUE THREAD

**Issue by funderburkjim** [sanskrit-lexicon/PWK#84]
We may at some later time wish to provide a linkable target for the citations in PWK  (`<ls>MBH. x,y,z</ls>`) to the so-called 
Bombay edition of Mahabharata.
@Andhrabharati may have sources for these if and when we undertake this. See [this comment](https://github.com/sanskrit-lexicon/COLOGNE/issues/371#issuecomment-1001463025).

**Comment by Andhrabharati** [comment]
Yes, I do have all those Bomb. ed. volumes (11 nos. of text + one index volume) and many more editions too!

**Comment by gasyoun** [comment]
> all those Bomb. ed. volumes

Located ...

**Comment by Andhrabharati** [comment]
located the Bomb. ed. volumes yourself?

Then you might consider putting someone in your team to index them.

**Comment by Andhrabharati** [comment]
Parva-1:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/35f4256c-718b-414b-9831-d8cbb5c91caf)

Parva-2:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/06177922-e871-4612-a364-1833acda9ec4)

Parva-3:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/7beb73b1-a445-4719-915c-d7f8844cd0a9)

Parva-4:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/d9123c16-6082-4901-9dad-3e8e436e1401)

Parva-5:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/2507a429-2b09-4681-b58f-a7560a4d5246)

Parva-6:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/b7fd8da2-4bcf-4186-8385-1983e4221450)

Parva-7:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/c94cca02-6421-47ec-9fd2-ef8fe66c9755)

Parva-8:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/25137a55-b57e-4bc5-b659-7c26d050bb8b)

Parva-9:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/4f2861a2-c4d3-4e91-a892-6602bee4dc5b)

Parva-10:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/48956c2e-3348-4980-a497-58c8c0ea4c35)

Parva-11:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/13e3083c-6254-44c3-9cae-c065dbbd7ccb)

Parva-13:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/34cdba57-be2a-440d-aec0-ede7c871a752)

Parva-14:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/0701eebf-4161-48e0-8dcc-f7186c9d0e87)

Parva-15:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/f3f7a5b7-8fc9-4836-95a6-10d8b14033aa)

Parva-16:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/5b7a017b-f2bb-42a3-9188-0a564bd7d04f)

Parva-17:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/4e077f99-9389-4d64-99ec-3da611aec9ff)

Parva-18:
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/365a3e4e-75b5-49cb-8be3-9ba79bcbf1b9)

**Comment by Andhrabharati** [comment]
While it is seen that PWG has very nominal (negligible) no. of citations from the Bomb. ed. of MBh. as compared to the Calc. ed., the pwk has exactly the opposite counts.

However, the citations in pwk are not that many (its 'theme' being mostly to be 'concise'), so the MBh. citations are just about 3.3K+ (against the count of 66K+ in PWG).

**So, is it worth putting effort to link these to pdf pages (the pages count about 4500!)?**

Now, I have a surprise Christmas gift to @funderburkjim / @gasyoun  [it being the 2nd anniv. of linking the MBh. (Calc. ed.) scans at CDSL now]--
[MBH-1863 Titular pages etc..pdf](https://github.com/sanskrit-lexicon/PWK/files/13761298/MBH-1863.Titular.pages.etc.pdf)

**Comment by Andhrabharati** [comment]
> located the Bomb. ed. volumes yourself?
> 
> Then you might consider putting someone in your team to index them.

@gasyoun 

You can get the Bomb. ed. MBh. from [this link](https://play.google.com/store/search?q=Mah%C3%A2bh%C3%A2rata+%C3%A9pop%C3%A9e+classique&c=books&pli=1)

I had renamed them at my end for ease of 'identification', as
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/e60dfd6a-c5ae-4d04-be87-980e726c215d)

[It was surprising to see that these were at a French Library, but not at any German or English Libraries!!]

Now the baton goes to you, to have these 'indexed'.

**Comment by gasyoun** [comment]
> Now the baton goes to you, to have these 'indexed'.

So what would give us about 4500 links, @Andhrabharati, right?

**Comment by Andhrabharati** [comment]
yes, a little (a hundred or two) above 4500 lines of index file to be made.

**Comment by funderburkjim** [comment]
> Now, I have a surprise Christmas gift ...

@Andhrabharati 

Thank you for the Christmas gift.  Very festive !

**Comment by funderburkjim** [comment]
> Re the Bomb. ed. MBh.  at Google play link above

* 6 volumes.  first volume has parva 1,2.
* When Viewed in edge browser, pages are sideways
  * Not a pdf but an ereader form.  
  * Is there a better way to view this on a desktop computer?
  
But I do think that a useable indexed version would be good to have somewhere.  The form I'm thinking of would be similar to the cdsl display of Kale Higher Sanskrit Grammar, or maybe similar to the Gorresio/Schlegel editions of Ramayana.

> So, is it worth putting effort to link these to pdf pages (the pages count about 4500!)

Good question.  But until there is a good link target, this question is moot.

**Comment by Andhrabharati** [comment]
> * Is there a better way to view this on a desktop computer?

Just "rotate right" in acrobat, not in a browser, and save the file first.

**Comment by Andhrabharati** [comment]
@funderburkjim 

It seems you do not have a facility to download the google books as PDFs; you may download these from my [Google drive](https://drive.google.com/drive/folders/1ENAswd1x5g3MO8Y6ZIkiPSLRI0pynIDq?usp=sharing).

**Comment by funderburkjim** [comment]
Downloaded from your share. Re vol. 1: rotate right works right.
When magnified, the Devanagari is quite clear -- might even be susceptible to ocr.

**Comment by Andhrabharati** [comment]
Yes, these books can be OCRed.

**Comment by Andhrabharati** [comment]
>>So, is it worth putting effort to link these to pdf pages (the pages count about 4500!)

>Good question. But until there is a good link target, this question is moot.

My point was that in the same that takes for this job, 4-5 smaller books could be indexed (each of which are having far more citations from them).

**Comment by gasyoun** [comment]
> 4-5 smaller books could be indexed (each of which are having far more citations from them).

@Andhrabharati we even had a list, somewhere of the more quoted ones...

**Comment by Andhrabharati** [comment]
I had done 4-5 already, but Jim is not anywhere around to take them and do the linking.

**Comment by Andhrabharati** [comment]
@gasyoun 

[This](https://github.com/sanskrit-lexicon/PWG/issues/51#issuecomment-1038467151) was your post that has the counts etc. in PWG (the counts have substantially changed over the next 2-3 months) and subsequently I had posted a couple beneath the same [and the works mentioned in those 4 posts were 'indexed' by me.]

These were the ones having citation counts **over 10K** (at that time), leaving **Pāṇini** and the Skt. lexicons [**AK.** (on which Jim seems to have done some work, but left incomplete), **H.** and **MED.**]; it may be noted that the **ŚKDr.** though comes far high above many, is not a source that could be linked as is [I do have the best possible Bengali edition volumes with me], as only the citations in it were cited in PWG, and not the entries/words of **ŚKDr.**

And about 6 months prior to that there was a [post](https://github.com/sanskrit-lexicon/PWG/issues/41#issuecomment-894007796) by @funderburkjim, to which I gave links for other works that he was asking.

**Comment by funderburkjim** [comment]
> I had done 4-5 already, but Jim is not anywhere around to take them and do the linking.

@Andhrabharati  Why don't you open a new issue, and provide one of your 4-5.

**Comment by Andhrabharati** [comment]
@funderburkjim 

I would rather suggest you to start with what is already 'opened' for you, that needs no indexing.
And, do you recall the same?

**Comment by Andhrabharati** [comment]
Then, I shall give the index files for the following works all at once, or one-by-one--

1. BHĀG. P. (Burnouf)
2. KATHĀS. (Brockhaus)
3. M. (Deslongchamps)
4. ŚAT. BR. (Weber)

**Comment by Andhrabharati** [comment]
Do you recall where you said thus?-

> From the last few days, there are 15+ Github posts from @Andhrabharati requesting my attention.
> The number of similar but older posts is unknown, but likely greater than 0.

**Comment by gasyoun** [comment]
>  it may be noted that the ŚKDr. though comes far high above many, is not a source that could be linked as is [I do have the best possible Bengali edition volumes with me], as only the citations in it were cited in PWG, and not the entries/words of ŚKDr.

@Andhrabharati still do not understand why can't be linked.

**Comment by Andhrabharati** [comment]
> > I had done 4-5 already, but Jim is not anywhere around to take them and do the linking.
> 
> @Andhrabharati Why don't you open a new issue, and provide one of your 4-5.


> Then, I shall give the index files for the following works all at once, or one-by-one--
>
>     1. BHĀG. P. (Burnouf)
>     2. KATHĀS. (Brockhaus)
>     3. M. (Deslongchamps)
>     4. ŚAT. BR. (Weber)

@funderburkjim 

Now that the **Verz. d. Oxf. H.** got linked up, I can post the index files for the other major works (that were already done by me), if and when you are ready by opening up a new issue, as you had suggested.

**Comment by funderburkjim** [comment]
@Andhrabharati Since this is on your mind,  why don't you send me the needed links or pdfs.  I'll do the downloads now to get local copies, and then attend to the necessary work to make link targets when time permits.

**Comment by Andhrabharati** [comment]
@funderburkjim 

I had hinted the KATHĀS. links [earlier](https://github.com/sanskrit-lexicon/PWG/issues/51#issuecomment-1041429823); probably you might start with the same.

**Comment by gasyoun** [comment]
@Andhrabharati how many done?

> I can post the index files for the other major works (that were already done by me)

**Comment by Andhrabharati** [comment]
https://github.com/sanskrit-lexicon/PWK/issues/84#issuecomment-1912322301_

**Comment by ramray** [comment]
[индексы для словаря МХБх Бобмбейское издание 1864 (indexes for the MXBx dictionary Bobmbay edition 1864).xlsx](https://github.com/user-attachments/files/18921408/1864.indexes.for.the.MXBx.dictionary.Bobmbay.edition.1864.xlsx)
First vol already
@funderburkjim

**Comment by angalinde** [comment]
[MBh_Bomb_vol_6_Index.txt](https://github.com/user-attachments/files/18931139/MBh_Bomb_vol_6_Index.txt)
[MBh_Bomb_vol_6_Comments.txt](https://github.com/user-attachments/files/18931138/MBh_Bomb_vol_6_Comments.txt)

**Comment by Azanuka2412** [comment]
[MBh_4_Index.xlsx](https://github.com/user-attachments/files/19274415/MBh_4_Index.xlsx)
[MBh_Bomb_vol_4_Comments.txt](https://github.com/user-attachments/files/19274414/MBh_Bomb_vol_4_Comments.txt)

ed. 1 — i-page numbers have been supplemented by letters
ed. 2 — pages with pictures have been added

**Comment by funderburkjim** [comment]
@Azanuka2412   @angalinde  
I don't understand the 'ed. 1' and 'ed. 2'   comments.  

Are you indexing the pdfs named in [this comment above](https://github.com/sanskrit-lexicon/PWK/issues/84#issuecomment-1893106243)?

**Comment by Andhrabharati** [comment]
Jim, they meant 'revision' to their index. They are for the above pdfs only.

**Comment by angalinde** [comment]
> [@Azanuka2412](https://github.com/Azanuka2412) [@angalinde](https://github.com/angalinde) I don't understand the 'ed. 1' and 'ed. 2' comments.
> 
> Are you indexing the pdfs named in [this comment above](https://github.com/sanskrit-lexicon/PWK/issues/84#issuecomment-1893106243)?

Yes that are the same pdfs. In process I slightly fixed the index and updated the attached file, the corrections are explained.

**Comment by Azanuka2412** [comment]
[Mbh_5_Index.xlsx](https://github.com/user-attachments/files/19394132/Mbh_5_Index.xlsx)
[MBh_Bomb_vol_5_Comments.txt](https://github.com/user-attachments/files/19394133/MBh_Bomb_vol_5_Comments.txt)

**Comment by funderburkjim** [comment]
We now have indexes for volumes 1,4,5,6.

**Comment by grigoriyt1** [comment]
@funderburkjim 

We have second volume done by Roman:

[MBh_Bomb_Vol_2_Index.xlsx](https://github.com/user-attachments/files/19486063/MBh_Bomb_Vol_2_Index.xlsx)

**Comment by grigoriyt1** [comment]
First volume. Uploading correct version from another repo (https://github.com/sanskrit-lexicon/PWG/issues/112#issuecomment-2700411279)
[Mahabharata_Bombay_Vol_1_Index (1).xlsx](https://github.com/user-attachments/files/19486162/Mahabharata_Bombay_Vol_1_Index.1.xlsx)

**Comment by funderburkjim** [comment]
@grigoriyt1 @Andhrabharati 

In the 6 index files,  there are 20 lines where an 'x' or 'y' occurs in the 'page' field.

[index_xy.txt](https://github.com/user-attachments/files/20092246/index_xy.txt)

Are these places where our pdfs are deficient?
Do you plan to add 'placeholder' pdf pages (such as 'mbh1-624x.pdf' ?)
Please explain.

**Comment by Andhrabharati** [comment]
> Are these places where our pdfs are deficient?

Yes, they denote the "missing pages" in the particular ed. scan copies.

> Do you plan to add 'placeholder' pdf pages (such as 'mbh1-624x.pdf' ) ?

And yes, we can think of adding the replacement pages wherever possible.

**Comment by Andhrabharati** [comment]
> [@Andhrabharati](https://github.com/Andhrabharati) Have I explained this properly?
> 
> Why are we doing a link target for Bombay edition of mahabharata, when there are so few links from koshas?

Yes, you have made a good enough try to put your explanation; but, there is a big catch that you have missed!

Almost all the citations [**4000+**] in pw(k) are from the Bomb. ed. only and we had talked about this earlier [1](https://github.com/sanskrit-lexicon/PWG/issues/48#issuecomment-1023881001), [2](https://github.com/sanskrit-lexicon/PWG/issues/48#issuecomment-1024830811), [3](); **probably this got skipped from your mind**!!

So are the MBh. citations from pwkvn and SCH that have 3 numeral parameters (x,y,z), **which are sufficiently high in numbers to be linked up**!

And I don't want to waste your time further, by explaining about "the less numbers in PWG and MW"!

**Comment by funderburkjim** [comment]
@grigoriyt1   ready for repo in sanskrit-lexicon-scans
Repo name  `mbhbomb`.  
suggest pdf file names like `mbhb-VNNNN.pdf `

**Comment by grigoriyt1** [comment]
repository is ready: **https://github.com/sanskrit-lexicon-scans/mbhbomb**

**Comment by gasyoun** [comment]
> Then I had resolved the 50 cases listed by you (47 certain, and 3 somewhat doubtful), apart from noting that one correction you did in the 19 cases has to be reverted.

Thanks

**Comment by funderburkjim** [comment]
## app0 and app1 constructed.
for index work, see files in [pwkissues/issue84](https://github.com/sanskrit-lexicon/PWK/tree/master/pwkissues/issue84)
* Using index_AB.txt
* includes my 19 changes and AB's 5 changes
* links to mbhbomb should work for pw, pwkvn, pwg.  
   * mw and sch will be constructed once pw work is done

Ready for @Andhrabharati to share corrections for the list of 50 from pw.

**Comment by Andhrabharati** [comment]
Here are 45 cases resolved out of 50--

[check_all_pw3_1_nopagerec (resolved).txt](https://github.com/user-attachments/files/20622834/check_all_pw3_1_nopagerec.resolved.txt)

**Comment by Andhrabharati** [comment]
Here are the other two cases resolved in pwk--

> 
> 002
> TODO key (1,127,72): pagerec not found
> L= 57387, hw= nAgadanta, pc=3-189-c
> check: ?
> AB: (1,128,72) ;; print_change

`----------------------------------------------`

> 046
> TODO key (13,150,45): pagerec not found
> L= 123561, hw= sAMKya, pc=7-109-a
> check: ?
> AB: (13,151,45) ;; print_change

`----------------------------------------------`

**Comment by Andhrabharati** [comment]
Isn't the "L= 69288, hw= pUrvendra, pc=4-115-b" corrected above?

**Comment by Andhrabharati** [comment]
And finally, here are the 3 remaining ones (**getting resolved**) in pwk--

> 016
> TODO key (2,79,32): pagerec not found
> L= 53258, hw= dyUtamaRqala, pc=3-126-c
> check: ?
> AB: (2,79,32) ;;page scan missed

`----------------------------------------------`

> 047
> TODO key (13,169,80): pagerec not found
> L= 111981, hw= SAs, pc=6-228-b
> check: ?
> AB: (3,168,80)  ;;print_change (has आशास्त, as per PWG [3,12071] citation)

`----------------------------------------------`

> 050
> TODO key (13,313,54): pagerec not found
> L= 209163, hw= prARa, pc=6-304-a
> check: ?
> AB: (3,313,54) ;;print_change

`----------------------------------------------`

**Comment by Andhrabharati** [comment]
Yet another correction in the index file--

> 1	664x	2	79	3	20	79b	remark=no scan
> 1	664y	2	79	21	27	80a	remark=no scan

to be corrected as

> 1	664x	2	79	3	22	79b	remark=no scan
> 1	664y	2	79	23	36	80a	remark=no scan

**Comment by Andhrabharati** [comment]
@funderburkjim 

This listing of "nopagerec" items is quite useful in bringing out the **problematic ones** in the citations.

Is this the first time that you had done this?
[I haven't noticed such list(s) for any other (**indexed &**) linked work so far.]

**Comment by funderburkjim** [comment]
```
016
TODO key (2,79,32): pagerec not found
L= 53258, hw= dyUtamaRqala, pc=3-126-c
check: ?
AB: (2,79,32) ;;page scan missed
```

iindex_AB, 2,79  matches verses 1-27.   but does  not match verse 32.
So some change needed either to index_AB or to the reference 2,79,32.

**Comment by funderburkjim** [comment]
I just saw your changes to index_AB --  that takes care of dyUtamaRqala !

**Comment by Andhrabharati** [comment]
> > Isn't the "L= 69288, hw= pUrvendra, pc=4-115-b" corrected above?
> 
> Further correction is needed
> 
> ```
> from change_pw2_pw3.txt:
> 285774 old {#pUrvendra#}¦ <lex>m.</lex> {%ein ehemaliger%} <is>Indra</is> 
> <ls>MBH. 1,191,27</ls>; <ab>vgl.</ab> <ls n="MBH. 1,191,">35</ls>.
> 
> 285774 new {#pUrvendra#}¦ <lex>m.</lex> {%ein ehemaliger%} <is>Indra</is> 
>   <ls>MBH. 1,196,27</ls>; <ab>vgl.</ab> <ls n="MBH. 1,196,">35</ls>.
> ```
> 
> The change Was made to pw.txt. However, the new refs 1,196,27 (and 1,196,35) are not found in index_AB.txt. So, it is still a 'pagerec not found'.

My bad here!
It should've been 197, and not 196 at these citations.

![Image](https://github.com/user-attachments/assets/ca184c7b-e30c-4301-a71c-a3eed3d16bcf)

![Image](https://github.com/user-attachments/assets/95af65b9-a633-402a-9c5b-ed65a3f9a156)

![Image](https://github.com/user-attachments/assets/eb97f549-f9dd-4efe-bc50-0f2cc4d0a705)

**Comment by funderburkjim** [comment]
> listing of "nopagerec" items is quite useful  
Is this the first time that you had done this?

The idea was thought of first in https://github.com/sanskrit-lexicon/PWG/issues/149 (MĀLAVIKĀGNIMITRA link target).
But this MBH Bombay is first real use.

**Comment by Andhrabharati** [comment]
So, why not apply this idea to all the works linked so far, to weed out such items?

**Comment by funderburkjim** [comment]
## pw 'pagerec not found' resolved
Working files [issue84](https://github.com/sanskrit-lexicon/PWK/tree/master/pwkissues/issue84)

* using index_AB1.txt (lines 935, 936 changed)
  * also used in app1
* 3 revisions to pw.txt (change_pw3_pw4.txt).
* check_all_pw3_3.txt  has all the instances of 3-parameter pw references linked to the index.

**Comment by funderburkjim** [comment]
## missing pages 
@Andhrabharati  you apparently have a source for missing pages of mbhbomb.

In index_AB1.txt, I count 33 such.

index_AB1_missing.txt  has adjusted the 'page' and 'remark' fields for these 33.  
* remark field  has the name of pdf file for given missing.  
   * This extends the naming convention used for non-missing pages.
* page field has the page with .1, .2, 3. convention..

This is better explained by comparing [index_AB1_missing.txt](https://github.com/user-attachments/files/20641276/index_AB1_missing.txt).

If you provide 33 pdf files so-named,  then I can integrate these into the link target.

**Comment by Andhrabharati** [comment]
Here are the snippets (**second batch**) for the resolutions from the "missed scan" pages--

> 025
> TODO key (4,72,22): pagerec not found
> L= 59846, hw= niSaWa, pc=3-222-c
> check: ?
> AB: (4,72,22) ;;page scan missed

![Image](https://github.com/user-attachments/assets/b513c074-e06e-4792-8345-7f9310e13f33)

> 029
> TODO key (7,195,11): pagerec not found
> L= 102888, hw= viDarmaka, pc=6-096-b
> check: ?
> AB: (7,195,11) ;;page scan missed

![Image](https://github.com/user-attachments/assets/17ae7bc1-c7da-4962-896a-3bafd2dae267)

> 031
> TODO key (8,56,103): pagerec not found
> L= 215960, hw= kAdraveya, pc=7-331-b
> check: ?
> AB: (8,56,103) ;;page scan missed

![Image](https://github.com/user-attachments/assets/5b0a5049-4419-437f-ad9b-58b4ef20f281)


034
TODO key (8,82,31): pagerec not found
L= 104497, hw= viS, pc=6-120-a
check: ?
AB: (8,82,31) ;;page scan missed (has अभिनिविष्ट, as in the pwk  citation)

![Image](https://github.com/user-attachments/assets/0baf113a-625e-446c-add2-59815dafa1c0)

**Comment by funderburkjim** [comment]
## almost finished 

Work files are [here](https://github.com/sanskrit-lexicon/PWK/tree/master/pwkissues/issue84).

changes to finish resolution of 'pagerec not found' condition; revisions to
* index_AB1.txt 
   * 1	95	1	11	1	19	42b   #  19 was 11
 * mw :   see change_mw_mw1.txt and change_mw1_mw2.txt 
 * pwkvn : see change_pwkvn_pwkvn1.txt
 * sch : see change_sch_sch1.txt

@Andhrabharati   Are we going to add the missing pages to mbhbomb repo (as mentioned [above](https://github.com/[funderburkjim](https://github.com/funderburkjim)) ?

If not, I think this issue can be closed.

**Comment by funderburkjim** [comment]
## missing pages installed

I think all is as it should be.

 Request others (@Andhrabharati  @grigoriyt1 etc) to do some checking before we close this issue.
